File: Dart/two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int, int> numToIndex = {};
    for (int i = 0; i < nums.length; i++) {
      int comploment = target - nums[i];
      if (numToIndex.containsKey(comploment)) {
        return [numToIndex[comploment]!, i];
      }
      numToIndex[nums[i]] = i;
    }
    return [-1, -1];
  }
}

void main() {
  Solution solution = Solution();
  List<int> result = solution.twoSum([3, 2, 4], 6);
  print(result);
}

File: Java/two-sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> dict = new HashMap<>();
        for(int i =0;i<nums.length;i++){
            int dif=target-nums[i];
            if(dict.containsKey(dif)){
                return new int[]{dict.get(dif),i};
            }
            dict.put(nums[i],i);
        }
        return new int[0];
    }
}

